File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
const go = (idx, v, target, numbers) => {
    if(idx === numbers.length) return v === target ? 1 : 0;
    
    
    return go(idx + 1, v + numbers[idx], target, numbers) + go(idx + 1, v - numbers[idx], target, numbers);
}

function solution(numbers, target) {
    return go(0, 0, target, numbers);
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

int go(int idx, int v, int target, vector<int> &numbers){
    // 기저사례
    if(idx == numbers.size()){
        if(v == target) return 1;
        else return 0;
    }
    
    // 로직
    return go(idx + 1, v + numbers[idx], target, numbers) + go(idx + 1, v - numbers[idx], target, numbers);
}

int solution(vector<int> numbers, int target) {    
    return go(0, 0, target, numbers);
}